### ISSUE TITLE
bcbio 1.1.2 SV calling with Lumpy and Smoove produces header only output and error in log

### ISSUE THREAD

**Issue by WimSpee** [bcbio/bcbio-nextgen#2628]
Hi,

Thank you (and also @brentp ) for including Lumpy with Smoove pre-processing in bcbio.

We got good results with Lumpy in the past , but were not able to run it within reasonable time on larger sets of samples. We really look forward to the results of Lumpy with Smoove pre-processing on larger sets of samples. 

I did a first analysis with the new Lumpy and Smoove using bcbio 1.1.2 .  The analysis finished but produced header only output VCF.gz files. 

Below is the error that can be found in the log file. There are specific line numbers of the Smoove code were the error was probably thrown. 

I SV called the same input data with Manta using bcbio 1.0.8. So the input data should be fine I think. 
I also tried deleting the structural directory and then restarting the analysis with 10 CPU on a single machine. This caused the same error. So my guess is that it is also not an out of resource error. 

Could you please have a look if you can identify and fix the error. Thank you. 

```
[smoove] 2019/01/15 16:54:20 removed 513434 alignments out of 3287232 (15.62%) that were bad interchromosomals or flanked-splitters from DA_575_01.split.bam
[smoove] 2019/01/15 16:54:20 removed 0 singletons out of 0 reads (NaN%) from DA_575_01.split.bam in 0 seconds
[smoove] 2019/01/15 16:54:23 removed 2396769 alignments out of 2863992 (83.69%) with depth > 800 or from excluded chroms from DA_575_03.split.bam in 23 seconds
[smoove] 2019/01/15 16:54:23 removed 467223 alignments out of 2863992 (16.31%) that were bad interchromosomals or flanked-splitters from DA_575_03.split.bam
[smoove] 2019/01/15 16:54:23 removed 0 singletons out of 0 reads (NaN%) from DA_575_03.split.bam in 0 seconds
[smoove] 2019/01/15 16:54:23 starting lumpy
[smoove] 2019/01/15 16:54:23 wrote lumpy command to /data/run/Projects/DA-1336/Mapping_SV_VariantCalling_V3_Lumpy/DA_1336_samples-merged/work/bcbiotx/tmpn2Fvs6/S_101913_04-DA_1336-svs-lumpy-cmd.sh
[smoove] 2019/01/15 16:54:23 writing sorted, indexed file to /data/run/Projects/DA-1336/Mapping_SV_VariantCalling_V3_Lumpy/DA_1336_samples-merged/work/bcbiotx/tmpn2Fvs6/S_101913_04-DA_1336-svs-smoove.genotyped.vcf.gz
[smoove] 2019/01/15 16:54:23 excluding variants with all unknown or homozygous reference genotypes
[smoove] 2019/01/15 16:54:24 > gsort version 0.0.6
[smoove] 2019/01/15 16:54:24 427
[smoove] 2019/01/15 16:54:24 0
[smoove] 2019/01/15 16:57:08 2019/01/15 16:57:08 EOF
[smoove] 2019/01/15 16:57:08 Failed to open -: unknown file type
[smoove] 2019/01/15 16:57:08 Failed to open -: unknown file type
[smoove] 2019/01/15 16:57:08 wrote sorted, indexed file to /data/run/Projects/DA-1336/DA_1336_samples-merged/work/bcbiotx/tmpn2Fvs6/S_101913_04-DA_1336-svs-smoove.genotyped.vcf.gz
panic: exit status 255
goroutine 1 [running]:
github.com/brentp/smoove/svtyper.check(0x9f3060, 0xc422040140)
        /home/brentp/go/src/github.com/brentp/smoove/svtyper/svtyper.go:32 +0x4a
github.com/brentp/smoove/svtyper.Svtyper(0x9f2b80, 0xc4215ba938, 0x7fff5db3ebd5, 0x99, 0xc4202c6200, 0x1b, 0x20, 0x7fff5db3ec9b, 0xa6, 0x7fff5db3ec76, ...)
        /home/brentp/go/src/github.com/brentp/smoove/svtyper/svtyper.go:215 +0xbee
github.com/brentp/smoove/lumpy.Main()
        /home/brentp/go/src/github.com/brentp/smoove/lumpy/lumpy.go:287 +0x3fb
main.main()
        /home/brentp/go/src/github.com/brentp/smoove/cmd/smoove/smoove.go:115 +0x1dd
' returned non-zero exit status 2
[2019-01-15T15:57Z] machine4v: bgzip S_101913_04-DA_1336-svs-smoove.genotyped.vcf
[2019-01-15T15:57Z] machine4v: tabix index S_101913_04-DA_1336-svs-smoove.genotyped.vcf.gz
[2019-01-15T15:57Z] machine4v: tabix index S_101913_04-DA_1336-svs-smoove.genotyped-filter.vcf.gz
[2019-01-15T15:57Z] machine4n: ipython: finalize_sv
```

**Comment by brentp** [comment]
The most informative line is here:
```
[smoove] 2019/01/15 16:54:20 removed 0 singletons out of 0 reads (NaN%) from DA_575_01.split.bam in 0 seconds
```
somehow, that part is getting no split reads. Are you sure you are not running out of disk space in $TMPDIR or in your output directory? And/or is there some NFS mount between you and the results dir?

**Comment by WimSpee** [comment]
Thank you for your fast answer.  I'll have a look at the (temporary) BAM files and the $TMPDIR.  
The `bcbio` work and `transaction` directory are on a HPC shared file system. I need to check if the $TMPDIR is also on the shared file system or on the HPC execution node it self. 

What I find strange is that there is a line just before the one you mention that shows that their are records in the same `DA_575_01.split.bam` file. With also the same time stamp `16:54:20`
```
[smoove] 2019/01/15 16:54:20 removed 513434 alignments out of 3287232 (15.62%) that were bad interchromosomals or flanked-splitters from DA_575_01.split.bam
[smoove] 2019/01/15 16:54:20 removed 0 singletons out of 0 reads (NaN%) from DA_575_01.split.bam in 0 seconds
```

**Comment by WimSpee** [comment]
I found the issue by trying to run the smoove command outside of bcbio. 
All the chomosomes of the assembly where added to the exclude chromosome argument.
```
  --excludechroms '~^GL,~^HLA,~_random,~^chrUn,~alt,~decoy,Chr_01,Chr_02,Chr_03,Chr_04,Chr_05,Chr_06,Chr_07,Chr_08,Chr_09,Chr_10,Chr_11,Chr_00' 
```

I edited the bcbio wrapper to not set the exclude chromosomes argument, and now Smoove runs past where it exited before. 

@chapmanb is there a way to whitelist custom Chr_XX chromosome names?

**Comment by brentp** [comment]
ah. ok. thanks for looking into that.

**Comment by WimSpee** [comment]
Thanks for the help. I could/should have looked into it before reaching out. Thank you for creating Smoove and improving the scalability of Lumpy.

**Comment by brentp** [comment]
sure thing. happy to hear any feedback.

@chapmanb it doesn't look like you are using duphold inside of smoove. You can just add `-d` to smoove genotype or smoove call and it will add depth info that's very useful for filtering.

**Comment by chapmanb** [comment]
Wim and Brent -- thanks so much for this report and for all the bug hunting. Apologies for not handling these different chromosome naming cases better. I pushed a fix to the latest development that should correctly identify the `Chr_01, Chr_02` cases as standard chromosomes and avoid excluding them. If you update `bcbio_nextgen.py upgrade -u development` and re-run it should hopefully work as expected.

Brent, right now we annotate with duphold as a secondary step after running smoove. I had integrated duphold prior to you adding this awesomeness to smoove, so need to go back and use it directly. Thanks for the heads up and push to adjust this.
